Packs/1Password/Integrations/OnePasswordEventCollector/OnePasswordEventCollector.py

```diff
@@ -0,0 +1,392 @@
+    """Checks if the bearer token has no access to any of the configured event types.
+
+    Args:
+        auth_introspection_response (dict): Response JSON from Client.introspect call.
+        event_types (list): List of event types from the integration configuration params.
+
+    Returns:
+        list[str]: List of unauthorized event types that the token does not have access to.
+    """
+    authorized_features = auth_introspection_response['features']
+    return [
+        event_type for event_type in event_types
+        if EVENT_TYPE_MAPPING.get(event_type) not in authorized_features
+    ]
+
+
+def add_fields_to_event(event: dict[str, Any], event_type: str):
+    """Sets the '_time' and 'event_type' fields to an event.
+
+    Args:
+        event (dict): Event dictionary with the new fields.
+        event_type (str): Type of 1Password event (e.g. 'Item usage actions', 'Sign in attempts').
+    """
+    event_time = arg_to_datetime(event['timestamp'], required=True)
+    event['_time'] = event_time.strftime(DATE_FORMAT)  # type: ignore[union-attr]
+    event['event_type'] = event_type
+
+
+def get_events_from_client(
+    client: Client,
+    event_type: str,
+    from_date: datetime,
+    max_events: int,
+    ids_to_skip: set[str] | None = None
+) -> list[dict]:
+    """Gets events of the specified type based on the 'from_date' filter and 'max_events' argument.
+
+    Args:
+        client (Client): 1Password Events API client.
+        event_type (str): Type of 1Password event (e.g. 'Item usage actions', 'Sign in attempts').
+        from_date (datetime): Datetime from which to get events.
+        max_events (int): The maximum number of events to fetch.
+        ids_to_skip (set | None): Optional set of already-fetched event UUIDs that should be skipped.
+
+    Returns:
+        list[dict]: List of events.
+    """
+    events: list[dict] = []
+    ids_to_skip = ids_to_skip or set()
+
+    # First call to the paginated API needs to to include a date filter
+    has_more_events = True
+    client_kwargs: dict[str, Any] = {'event_type': event_type, 'from_date': from_date}
+
+    while has_more_events:
+        response = client.get_events(**client_kwargs)
+        pagination_cursor = response['cursor']
+
+        for event in response['items']:
+            event_id = event['uuid']
+
+            if event_id in ids_to_skip:
+                continue
+
+            if len(events) == max_events:
+                break
+
+            add_fields_to_event(event, event_type)
+            events.append(event)
+            ids_to_skip.add(event_id)
+
+        demisto.debug(
+            f'Response has {len(response["items"])} events and pagination cursor: {pagination_cursor}.'
+            f'Used client arguments: {client_kwargs}.'
+        )
+
+        # Followup API calls need to include pagination cursor (if any more events)
+        has_more_events = False if len(events) == max_events else response['has_more']
```

Comment:
this is dangerous - what if there are still events in the server side but we limit to `max_events`? Will we surely get the correct next ones on the next run?

Reply:
Each fetch run starts by using a "**reset cursor**" (date filtering) for the first request and then uses "**pagination / continuing cursor**" in subsequent requests to navigate to the next page(s).

Even if we fetch part of the results on a given page and stop because we reached `max_events`, we can still re-access the data again in the next fetch run thanks to our use of the reset cursor.

You can refer to the documentation on the [1Password Events API Reset Cursor](https://developer.1password.com/docs/events-api/reference/#reset-cursor) for more information. Here is a small quote for connivence:

> _You can also use a reset cursor any time you need to reset the parameters of your cursor – such as the number of records to return with each request – or **go back to an earlier point in the records**._

Reply:
I have added a note in the function doc string and documented this in Confluence.

Following our in-person discussion, I am marking this as "Resolved". Feel free to re-open if needed.